### PR TITLE
docs(test): clarify docker usage for windows

### PR DESCRIPTION
**Pull request by liamdebeasi**
Team members on Windows were having trouble running tests inside of Docker. Given that headed tests must use WSL, I think it makes sense to have Windows devs use WSL for any E2E tests run inside of Docker. That way they aren't switching back and forth between shells.

This PR clarifies the Docker docs that instruct devs on how to use WSL.

### CODE REVIEW

File: core/src/utils/test/playwright/docs/usage-instructions.md [hunk 1]
```diff
@@ -32,6 +32,19 @@ The [Running Tests](#running-tests) and [Managing Screenshots](#managing-screens
 
 Docker can be installed by [following the steps on the Docker website](https://docs.docker.com/get-docker/).
 
+### Docker and Windows Development
+
+Developers using Windows who wish to run tests using Docker must use the [Windows Subsystem for Linux (WSL)](https://learn.microsoft.com/en-us/windows/wsl/about). Developers who wish to run headed tests will also need to use WSLg.
+
+If you are running Docker Desktop on Windows 10 or 11 you likely already have both WSL and WSLg installed. The following steps show how to verify that WSL and WSLg are installed. If either of the below verification checks fail, then developers should [download the latest version of WSL](https://apps.microsoft.com/store/detail/9P9TQF7MRM4R?hl=en-us&gl=US).
```
Comment (amandaejohnston) [inline core/src/utils/test/playwright/docs/usage-instructions.md:39]: Some other info that may be good to add: when I launched WSL, it opened, but then crashed/closed immediately. After some troubleshooting I learned that I didn't have Ubuntu installed, and had to do the following:
1. Run `wsl --install` in a terminal to install Ubuntu (I learned that it wasn't installed by running `wsl --list`, the result of which didn't include `Ubuntu`)
2. Run `exit` to quit out of WSL after it was installed and auto-launched in the terminal
3. Run `wsl --set-default Ubuntu`; my default distro had been set to a Docker one, again learned through `wsl --list` which gave the below result:
```
docker-desktop-data (Default)
Ubuntu
docker-desktop
```

Once that was all done, I was able to run WSL successfully and confirmed I had WSLg installed. The WSL terminal also started showing up in VSCode, where I didn't have the option previously. Unsure how typical my situation is though.

**Comment by brandyscarney** [comment]
Removing myself as a reviewer as this is being reviewed, but feel free to re-request if you need me to look at it.

File: core/src/utils/test/playwright/docs/usage-instructions.md (hunk 1 above)
Comment (liamdebeasi) [reply core/src/utils/test/playwright/docs/usage-instructions.md]: Good call out. Added in https://github.com/ionic-team/ionic-framework/pull/29274/commits/fe87310034a8b43949cd90907ad32c3ce8839342

**Review by amandaejohnston — APPROVED** [review]
Thank you!!

**Pull request merged**